Feature : Gemini Integration in chat

vaibhav-yerkar opened the pull request:
Fixes: #18 

## 📋 Description

Implemented Gemini Integration in chat 

## 🔨 Changes Made

- Implemented backend for gemini api request and response handling
- added new api route /api/command
- added new controller commandController and router commandRouter in backend/routes 
- modified frontend/src/component/SingleChat to handle api calling and command overlay display : so when user input '/' an overlay with available commands are displayed
- added new component in /src/component/miscellaneous CommandOvelay to display the available command

## 🔨 Screen Shots OR Short videos of working
https://github.com/user-attachments/assets/3694db1d-2aed-4410-ac27-7f6bc3bfb433


## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] Read Contributing.md

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [x] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## 🤝 Related Issues


## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to Talk-Trove! We look forward to reviewing your PR and getting it merged!*

kartikayasijaa committed:
Update README.md

kartikayasijaa committed:
Merge pull request #141 from kartikayasijaa/kartikayasijaa-patch-1

Update README.md

vaibhav-yerkar committed:
Implemented Gemini Integration to the chat

vaibhav-yerkar commented:
@kartikayasijaa , can you review my PR

kartikayasijaa commented on backend/controllers/commandControllers.js:
```diff
@@ -0,0 +1,128 @@
+
+dotenv.config()
+
+// const baseURL = "https://api.aimlapi.com/v1";
```
can we separate this to a new file

do like this 
integrations
 - chatgpt
 - gemini

vaibhav-yerkar replied:
Ok , sure

vaibhav-yerkar committed:
Modularise the command-Controller

vaibhav-yerkar commented:
added a folder commandControllers in the controller folder
added separate file for :
- chatgpt controller
- gemini controller

@kartikayasijaa

vaibhav-yerkar committed:
rename variable with corrected name

vaibhav-yerkar committed:
Merge branch 'test' into issue-18

vaibhav-yerkar commented:
@kartikayasijaa  can you review the changes

vaibhav-yerkar commented:
@kartikayasijaa its been a long time, can you please review my PR
